tests/e2e/cmp_test.go

```diff
@@ -0,0 +1,370 @@
+// +build !race
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"time"
+)
+
+// SourceTest checks the component-source-type and the source url in the annotation of the bc and dc
+// appTestName is the app of the app
+// sourceType is the type of the source of the component i.e git/binary/local
+// source is the source of the component i.e gitUrl or path to the directory or binary file
+func SourceTest(appTestName string, sourceType string, source string) {
+	// checking for source-type in dc
+	getDc := runCmd("oc get dc wildfly-" + appTestName + " -o go-template='{{index .metadata.annotations \"app.kubernetes.io/component-source-type\"}}'")
```

Comment:
can we extract out wildfly or the component type also to parameter in all the lines in this func?

Reply:
@anmolbabu this is an old function. I just did some rearrangement of tests with this PR. Let me see if I can change that. I agree with you, it would be nice to do that.